sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/ParameterAssignedToBase.cs

```diff
@@ -54,18 +72,52 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                         c.ReportDiagnosticWhenActive(Diagnostic.Create(SupportedDiagnostics[0], left.GetLocation(), left.ToString()));
                     }
                 },
-                SyntaxKindsOfInterest.ToArray());
+                SyntaxKindOfInterest);
         }
 
+        private bool IsReadBefore(SemanticModel semanticModel, ISymbol parameterSymbol, TAssignmentStatementSyntax assignment)
+        {
+            //FIXME: REMOVE DEBUG
+            if (parameterSymbol.Name.ToString() == "xx")
+                System.Diagnostics.Debugger.Break();
 
-        protected abstract bool IsAssignmentToCatchVariable(ISymbol symbol, SyntaxNode node);
 
-        protected abstract bool IsAssignmentToParameter(ISymbol symbol);
+            // Same problem as in VB.NET / IsAssignmentToCatchVariable:
+            // parameterSymbol.DeclaringSyntaxReferences is empty for Catch syntax in VB.NET as well as for indexer syntax for C#
+            // https://github.com/dotnet/roslyn/issues/6209
+            var stopLocation = parameterSymbol.Locations.FirstOrDefault();
+            if (stopLocation == null)
+            {
+                return true; //If we can't find the location, it's going to be FN
+            }
+            return GetPreviousNodes(stopLocation, assignment)
+                .Union(AssignmentRight(assignment).DescendantNodes())
+                .OfType<TIdentifierNameSyntax>()
+                .Any(node =>
+                {
+                    var nodeSymbol = semanticModel.GetSymbolInfo(node).Symbol;
+                    return parameterSymbol.Equals(nodeSymbol);
+                });
+        }
 
-        protected abstract bool IsReadBefore(SemanticModel semanticModel, ISymbol parameterSymbol, TAssignmentStatementSyntax assignment);
+        /// <summary>
+        /// Returns all nodes before the specified statement to the declaration of variable/parameter given by stopLocation.
+        /// This method recursively traverses all parent blocks of the provided statement.
+        /// </summary>
+        private static IEnumerable<SyntaxNode> GetPreviousNodes(Location stopLocation, SyntaxNode statement) 
+        {
+            if (statement == null || (stopLocation != null && statement.GetLocation().SourceSpan.IntersectsWith(stopLocation.SourceSpan)))   //Method declaration or Catch variable declaration, stop here and do not include this statement
```

Comment:
This null check: `stopLocation != null` is unnecessary here, as it is already checked for null before calling this method